### PR TITLE
Run4-hgx306 Backport the bugfix to HGCal geometry from the PR's 37330 and 37460

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Backport the bugfix to HGCal geometry from the PR's 37330 and 37460 which calculates the corners of the cells in the rotated layers correctly

#### PR validation:

Use the runTheMatrix test workflows and some code in the test area of Geometry/HGCa;Geometry

.#### if this PR is a backport please specify the original PR and why you need to backport that PR:

The PR #37330 is merged with the master. The PR #37460 only takes care of comments suggested by Andreas while merging #37330

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

This is a backport PR. It should be useful for Phase-2 production for L1-HLT.

**Comment by perrotta** [comment]
backport of https://github.com/cms-sw/cmssw/pull/37330

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include "Geometry/HGCalCommonData/interface/HGCalGeomParameters.h"
 #include "Geometry/HGCalCommonData/interface/HGCalGeomTools.h"
 #include "Geometry/HGCalCommonData/interface/HGCalGeometryMode.h"
+#include "Geometry/HGCalCommonData/interface/HGCalCell.h"
```
Comment (perrotta) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:13]: This addition is not needed, unless you also backport #37374

**Comment by perrotta** [comment]
@bsunanda is it not needed in the backport the test code and scripts which were merged with #37330?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by perrotta** [comment]
@srimanob you already signed this once, I think latest additions (a couple of test files, as in the master) should be rather innocuos: we are waiting anyhow for your further signature before merging this backport in 12_3_X

**Comment by srimanob** [comment]
+Upgrade

This is a backport PR which should be useful for comping production of Phase-2.

**Pull request merged**